### PR TITLE
Search `RzAnalysisOp.analysis_vals[]` for xrefs.

### PR DESCRIPTION
**Pull request by Rot127**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

The function `rz_core_analysis_search_xrefs` does not search through the `RzAnalysisOp.anlysis_vals[]` array, which was added in https://github.com/rizinorg/rizin/pull/1614.

In this PR a simple loop is added which searches each value for references.

**Test plan**

Test case was added: The `memw(R17+#0x0) = ##0x1c` instruction has the `0x1c` value stored in `RzAnalysisOp.analysis_vals[0]`.

**Closing issues**

None

### CODE REVIEW

File: librz/core/canalysis.c [hunk 1]
```diff
@@ -4223,6 +4223,16 @@ static bool found_xref(RzCore *core, ut64 at, ut64 xref_to, RzAnalysisXRefType t
...
+ * \brief Searches for xrefs in the range of the paramters \p 'from' and \p 'to'.
+ *
+ * \param core The Rizin core.
+ * \param from Start of search interval.
+ * \param to End of search interval.
+ * \param pj The print JSON object.
+ * \param rad The print mode. If set to 'j' the \p 'pj' parameter will filled with the xrefs found.
+ * \return int Number of found xrefs. -1 in case of failure.
+ */
 RZ_API int rz_core_analysis_search_xrefs(RzCore *core, ut64 from, ut64 to, PJ *pj, int rad) {
```
Comment (XVilka) [inline librz/core/canalysis.c]: While you are at it, could you please also change the `rad` parameter to use the enum? See https://github.com/rizinorg/rizin/issues/489

File: librz/core/canalysis.c [hunk 2]
```diff
@@ -4223,6 +4223,16 @@ static bool found_xref(RzCore *core, ut64 at, ut64 xref_to, RzAnalysisXRefType t
...
 	bool cfg_debug = rz_config_get_b(core->config, "cfg.debug");
 	bool cfg_analysis_strings = rz_config_get_i(core->config, "analysis.strings");
```
Comment (XVilka) [inline librz/core/canalysis.c:3390]: Also, I recommend splitting this function in two: one to perform the actual search, and store the result, and another to print the result.

Comment (XVilka) [reply librz/core/canalysis.c:3390]: Note, it might require changing also the `found_xref()`.

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/core/cmd/cmd_analysis.c [hunk 3]
```diff
@@ -4832,24 +4832,32 @@ static void cmd_analysis_graph(RzCore *core, const char *input) {
 RZ_API int rz_core_analysis_refs(RzCore *core, const char *input) {
 	int cfg_debug = rz_config_get_b(core->config, "cfg.debug");
 	ut64 from, to;
-	int rad;
+	RzOutputMode out_mode;
```
Comment (XVilka) [inline librz/core/cmd/cmd_analysis.c:4835]: Could you extract this particular change, also maybe port `aar` command to the rzshell (https://github.com/rizinorg/rizin/issues/1442) in a separate PR?

Comment (Rot127) [reply librz/core/cmd/cmd_analysis.c:4835]: With extracting you mean as a separated commit?

**Review by wargio — APPROVED** [review]

**Review by thestr4ng3r — APPROVED** [review]

**Pull request merged**